src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java

```diff
@@ -1020,19 +1047,21 @@ private boolean isValidUsersSession(final Map<String, String> sessionInformation
      *            - Current date
      * @param sessionToken
      *            - a token allowing session invalidation
-     * @param partialLoginFlag
+     * @param caveatFlags
      *            - Boolean data to encode in the cookie - true if a partial login
      * @return HMAC signature.
      */
     private String calculateSessionHMAC(final String key, final String userId, final String currentDate, final String sessionToken,
-                                        @Nullable final String partialLoginFlag) {
+                                        @Nullable final ArrayList<String> caveatFlags) {
         StringBuilder sb = new StringBuilder();
         sb.append(userId);
         sb.append("|").append(currentDate);
         sb.append("|").append(sessionToken);
 
-        if (partialLoginFlag != null) {
-            sb.append("|").append(partialLoginFlag);
+        if (null != caveatFlags) {
+            for (String c : caveatFlags) {
```

Comment:
This List was generated from a Set on at least one code path on the way here; it will not recognise two identical sets of caveats if the order happens to be different. If we're going to allow multiple caveats by using a List/Set/Collection, we should do so in a future-proof way.

Perhaps sorting the `caveatFlags` collection just before this loop is simplest and most effective?